src/main/java/org/springframework/integration/dsl/core/PollerSpec.java

```diff
@@ -79,6 +81,21 @@ public PollerSpec errorHandler(ErrorHandler errorHandler) {
 	}
 
 	/**
+	 * Specify a {@link MessageChannel} to use for sending error message in case
+	 * of polling failures.
+	 * @param errorChannel the {@link MessageChannel} to use.
+	 * @return the spec.
+	 * @since 1.2
+	 * @see MessagePublishingErrorHandler
+	 */
+	public PollerSpec errorChannel(MessageChannel errorChannel) {
+		MessagePublishingErrorHandler errorHandler = new MessagePublishingErrorHandler();
+		errorHandler.setDefaultErrorChannel(errorChannel);
+		this.componentToRegister.add(errorHandler);
+		return errorHandler(errorHandler);
+	}
+
```

Comment:
Correct. We need that fix first.
I think we can live some time with just only `MessageChannel` variant. 
Much better anyway!